_posts/2022-08-14-data-pipelines-production-ready-great-expectations.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title: "Make Your Trino Data Pipelines Production Ready Using Great Expectations"
+author: "Brian Olsen, Brian Zhan"
+excerpt_separator: <!--more-->
+---
+
+Great Expectations is an open source Python library that allows you to write
+detailed tests called
+[expectations](https://docs.greatexpectations.io/docs/terms/expectation/)
+against your data. Users write these expectations to run validations against the
+data as it enters your system. These expectations are expressed as methods in
+Python and stored in JSON and YAML files. One great advantage of expectations is
+the human readable documentation that results from these tests. As you role out
+different versions of the code, you get alerted to any unexpected changes and 
+have version specific generated documentation for what changed. In this blog, 
+we'll show you how to write expectations on tables in Trino!
+
+![](/assets/blog/data-pipelines-production-ready-great-expectations/trino-ge.png)
+
+
+<!--more-->
+
+## The need for data quality
+
+Managing data pipelines is not for the faint of heart. Nodes fail, you run
+out of memory, bursty traffic causes abnormal behavior, and that’s just the tip
+of the Apache Iceberg 😉. Lots of Trino community members build sophisticated
+data pipelines and data applications using Trino. This received a turbo-boost 
+when Trino added a 
+[fault tolerance]({% post_url 2022-05-05-tardigrade-launch %}) mode to safeguard
+against some of these failures when executing long-running and 
+resource-intensive queries. 
+
+Aside from all the infrastructure problems that concern data teams, another
+category of problems that has been the silent killer for quite some time is data
+quality. Faulty data comes in, which can either cause data pipelines to fail, or
+possibly go unnoticed and cause inaccurate downstream reporting. Tacit knowledge
+is scattered among domain experts, technical experts, and the code and data
+itself. Maintenance becomes time-consuming and expensive. Documentation gets out
+of date and unreliable. This is why using data quality checks using libraries 
+like [Great Expectations](https://greatexpectations.io/) is so important when
+writing ETL applications.
+
+## Improve data quality in Trino with Great Expectations
+
+As data quality becomes more in the forefront of the Trino community, the Great
```

Comment:
```suggestion
As data quality moves to the forefront of the Trino community, the Great
```

Reply:
I'm not sure something can be _more_ forefront-y, and this wording is hard to parse. If we want to emphasize "it was already really important, but now it's basically the most important," it might take a reword to make it clear that that's the point being made.

Reply:
yup bad grammar :)